tests/testthat/test-connection.R

```diff
@@ -89,39 +81,37 @@ test_that("is_token_expired() works for an expired and unexpired tokens", {
   new_payload <- list(user_id = 3, username = "model_owner1", exp = ten_min_from_now, email = "")
   new_payload_json <- jsonlite::toJSON(new_payload, auto_unbox = TRUE)
   unexpired_token <- paste0(old_header, '.', base64url::base64_urlencode(new_payload_json), '.', old_signature)
-
-  mock_authenticate(zoltar_connection, token = unexpired_token)
+  local_mocked_bindings(get_token = function(zoltar_session) { unexpired_token })
+  zoltar_authenticate(zoltar_connection, "username", "password")
   expect_false(is_token_expired(zoltar_connection$session))
 })
 
 
 test_that("get_resource() calls re_authenticate_if_necessary()", {
   zoltar_connection <- new_connection("http://example.com")
   m <- mock()
-  testthat::with_mock("zoltr::re_authenticate_if_necessary" = m, {
-    webmockr::stub_request("get", uri = "http://example.com/api/model/0/") %>%
-      to_return(
-        body = list(),
-        status = 200,
-        headers = list('Content-Type' = 'application/json; charset=utf-8'))
-    get_resource(zoltar_connection, "http://example.com/api/model/0/")
-    expect_equal(length(mock_calls(m)), 1)
-  })
+  local_mocked_bindings(re_authenticate_if_necessary = m)
+  webmockr::stub_request("get", uri = "http://example.com/api/model/0/") %>%
+    to_return(
+      body = list(),
+      status = 200,
+      headers = list('Content-Type' = 'application/json; charset=utf-8'))
+  get_resource(zoltar_connection, "http://example.com/api/model/0/")
+  expect_equal(length(mock_calls(m)), 1)
```

Comment:
Here's an alternative approach to using `mockery::mock()`. Instead of creating a mock object we use local binding to issue a message when the function is called and test for the message:
```r
  zoltar_connection <- new_connection("http://example.com")
  m <- function(...) {
    message("re_authenticate_if_necessary() called")
  }
  local_mocked_bindings(re_authenticate_if_necessary = m)
  webmockr::stub_request("get", uri = "http://example.com/api/model/0/") %>%
    webmockr::to_return(
      body = list(),
      status = 200,
      headers = list('Content-Type' = 'application/json; charset=utf-8'))
  expect_message(get_resource(zoltar_connection, "http://example.com/api/model/0/"),
                 regexp = "(re_authenticate_if_necessary\\(\\) called){1}")
```